content/news/posts/2025/breaking-change-guzzle-removal-v9.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title:  "Breaking Change in V9 Guzzle removal, what are the alternatives?"
+subtitle: "You are using Guzzle in your module, starting from PrestaShop V9 the library is not available by default any more, how can you adapt your module?"
+date:   2025-03-17
+authors: [ JonathanLelievre ]
+icon: icon-code
+tags: [autoupgrade, module, beta, releases, community, contribution, development, qa]
+---
+
+The PrestaShop V9 version is a major version, as such it comes with breaking changes among which some removed dependencies, you can find more details about those changes on the [developer documentation](https://devdocs.prestashop-project.org/9/modules/core-updates/9.0/), but in this article we're going to focus on one in particular the removal of Guzzle and how to adapt your module using it.
+
+## Why remove Guzzle in the first place?
+
+Guzzle has been one of the many dependencies included in the PrestaShop core for many years, as such modules were used to have it available and depended on it to make external HTTP requests.
+
+But this library has been replaced over the years by other tools in the core, most the Symfony HTTP Client. The core itself no longer had needs for this dependency, so we decided it was time to make the core lighter and remove guzzle along with other unused libraries.
+
+The weight of the build is not the only reason, in PrestaShop 8 we had to update Guzzle from version 5 to 7, this already created some incompatibilities at the time because of the breaking changes from Guzzle itself and modules struggled to maintain a cross compatibility between PrestaShop versions.
```

Comment:
```suggestion
The weight of the build is not the only reason. In PrestaShop 8 we had to update Guzzle from version 5 to 7, this already created some incompatibilities at the time because of the breaking changes from Guzzle itself and modules struggled to maintain a cross compatibility between PrestaShop versions.
```